network-rfc/12_contracts_design.md

```diff
@@ -0,0 +1,90 @@
+# Portal System
+
+
+### Portal Creation and SQD Collection
+
+The Portal System begins when a Data Consumer creates a new portal through the PortalFactory by specifying key parameters including the target amount of SQD tokens needed, deposit deadline, payment token, "budget".
+
+The data consumer allocation (contribution by the deployer) will be determined by the target amount that the data consumer is seeking.
```

Comment:
Data consumer is actually not the one who operates the portal. Let's try to stick to the already established terminology:
- **SQD provider** delegates SQD to earn USDC
- **Portal operator** opens a pool to collect delegated SQD to get a permission to run the portal, then pays USDC fees through the contract to SQD providers
- **Portal client** (actual data consumer) uses the portal hosted by portal operator and buys subscription provided by the operator (usually in fiat)